web/src/typing.ts

```diff
@@ -71,13 +95,32 @@ function send_typing_notification_based_on_message_type(
     to: Recipient,
     operation: "start" | "stop",
 ): void {
+    assert(to.notification_event_type === "typing");
     if (to.message_type === "direct" && user_settings.send_private_typing_notifications) {
         send_direct_message_typing_notification(to.ids, operation);
     } else if (to.message_type === "stream" && user_settings.send_stream_typing_notifications) {
         send_stream_typing_notification(to.stream_id, to.topic, operation);
     }
 }
 
+function send_typing_notifications_for_message_edit(
+    message_id: number,
+    operation: "start" | "stop",
+): void {
+    if (message_edit_typing_notifications_enabled(message_id)) {
+        send_message_edit_typing_notification_ajax(message_id, operation);
+    }
+}
+
+function message_edit_typing_notifications_enabled(message_id: number): boolean {
+    const message = message_store.get(message_id);
+    assert(message !== undefined);
+    if (message.type === "stream") {
+        return user_settings.send_stream_typing_notifications;
+    }
+    return user_settings.send_private_typing_notifications;
+}
```

Comment:
Define this before the function that uses it, rather than just after; it's more readable that way.